packages/aws-cdk/lib/api/util/sdk_ini_file.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * A reimplementation of JS AWS SDK's SharedIniFile class
+ *
+ * We need that class to parse the ~/.aws/config file to determine the correct
+ * region at runtime, but unfortunately it is private upstream.
```

Comment:
I definitely wanted to open an issue, but I'd like to land this first so that I can point them to the code and say "see, this is what we've had to do to work around the API deficiencies".

Pointing to code is easier than describing :)